File: C#Integration/PokerProtocol/Commands/Game/PlayerTurnBeganCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using EricUtility;
using PokerProtocol.Commands.Lobby.Response;
using PokerWorld.Game;
using EricUtility.Networking.Commands;

namespace PokerProtocol.Commands.Game
{
    public class PlayerTurnBeganCommand : AbstractCommand
    {
        protected override string CommandName
        {
            get { return PlayerTurnBeganCommand.COMMAND_NAME; }
        }
        public static string COMMAND_NAME = "gamePLAYER_TURN_BEGAN";

        private readonly int m_PlayerPos;
        private readonly int m_LastPlayerNoSeat;

        public int PlayerPos
        {
            get { return m_PlayerPos; }
        }
        public int LastPlayerNoSeat
        {
            get { return m_LastPlayerNoSeat; }
        }

        public PlayerTurnBeganCommand(StringTokenizer argsToken)
        {
            m_PlayerPos = int.Parse(argsToken.NextToken());
            m_LastPlayerNoSeat = int.Parse(argsToken.NextToken());
        }

        public PlayerTurnBeganCommand(int pos, int last)
        {
            m_PlayerPos = pos;
            m_LastPlayerNoSeat = last;
        }

        public override void Encode(StringBuilder sb)
        {
            Append(sb, m_PlayerPos);
            Append(sb, m_LastPlayerNoSeat);
        }
    }
}

File: trunk/JAVA/PokerProtocol/src/bluffinmuffin/protocol/commands2/game/PlayerTurnBeganCommand.java
package bluffinmuffin.protocol.commands2.game;

import java.util.StringTokenizer;

import bluffinmuffin.protocol.commands2.AbstractCommand;

public class PlayerTurnBeganCommand extends AbstractCommand
{
    @Override
    protected String getCommandName()
    {
        return PlayerTurnBeganCommand.COMMAND_NAME;
    }
    
    private final int m_playerPos;
    private final int m_lastPlayerNoSeat;
    
    public static String COMMAND_NAME = "gamePLAYER_TURN_BEGAN";
    
    public PlayerTurnBeganCommand(StringTokenizer argsToken)
    {
        m_playerPos = Integer.parseInt(argsToken.nextToken());
        m_lastPlayerNoSeat = Integer.parseInt(argsToken.nextToken());
    }
    
    public PlayerTurnBeganCommand(int pos, int lastNoSeat)
    {
        m_playerPos = pos;
        m_lastPlayerNoSeat = lastNoSeat;
    }
    
    @Override
    public void encode(StringBuilder sb)
    {
        append(sb, m_playerPos);
        append(sb, m_lastPlayerNoSeat);
    }
    
    public int getPlayerPos()
    {
        return m_playerPos;
    }
    
    public int getLastPlayerNoSeat()
    {
        return m_lastPlayerNoSeat;
    }
}
